src/Icon.tsx

```diff
@@ -137,7 +137,8 @@ Icon.propTypes = {
   ]),
   style: PropTypes.object,
   inStack: PropTypes.bool,
-  className: PropTypes.string
+  className: PropTypes.string,
+  onClick: PropTypes.func,

```

Comment:
All the other changes are fine, but I don't think we should encourage click handlers on the icon itself. These should always be applied to the wrapping component/element.